### ISSUE TITLE
Dealing with bad posts

### ISSUE THREAD

**Issue by zzyyxxww** [ranterix/drafts#2]
This is one of the most brought up points of devRants' problems.

Trying to define a bad post could deserve its own separate document, but we can make a quick approximation initially. A post is bad when a) it is posted in the wrong category b) it has extremely terrible gramar or syntax errors.

I think initially bad posts will be a non issue since the community would be very small so we should use a very simple mecanism to rely on downvotes to filter out bad posts.

I guess the most sensible would be hiding posts with -2 scores (which I think is how devrant works, correct me if I'm wrong). Hiding on just -1 would be too much power for everyone.

We could also give certain users that we can trust the power to hide posts on one downvote, having a record as well for when that happens.

Looking forward we will need a better solution to this. We could explore how other sites deal with this. For example, **hacker news** intentionally scares "low tier" users by having a very sober ui. But let's not get too deep into this yet.

**Comment by dudeking** [comment]
Main issue with devrant is the bad ui design that makes it easy to neglect the correct sections. The rant button takes up 95% of the screen, while everything else is tucked away. There should at least be equal space for each button. The bad grammar usually comes from foreigners, and the reasonable way to approach this is to perhaps make an algorithm that prefers the users language over the other languages, which separates the site into groups of the same language, but also allow users to view and write in English or any other language. Post hiding shouldn't be a thing, and should be yet another setting on an algorithm that pushes it up or down the feed.

That's the theory though

**Comment by zzyyxxww** [comment]
> Main issue with devrant is the bad ui design that makes it easy to neglect the correct sections. The rant button takes up 95% of the screen, while everything else is tucked away. There should at least be equal space for each button.

I agree, it is too big, let's not have that. I think using a dropdown will achieve that and will also be easy to implement. 

I'm considering the idea of making the joke/meme the default button. The reason for this is that if you're "baduser" enough to not bother to switch type, then users that have joke/memes deactivated will never hear from users like you.  
But I think a problem of that would be that it might seem we're a joke/meme centric community.

> The bad grammar usually comes from foreigners, and the reasonable way to approach this is to perhaps make an algorithm that prefers the users language over the other languages, which separates the site into groups of the same language, but also allow users to view and write in English or any other language.

I think we should definitely implement this idea at some point, but not initially so it's too difficult to take off. At some point we should do it since it would create more communication between non english and english speakers.

> Post hiding shouldn't be a thing, and should be yet another setting on an algorithm that pushes it up or down the feed, depending on the %of people who ++d and --d

I have mixed feelings about both having -- and not having --.
